src/server/main_service.cc

```diff
@@ -971,6 +971,8 @@ void Service::DispatchMC(const MemcacheParser::Command& cmd, std::string_view va
     }
   }
 
+  dfly_cntx->conn_state.memcache_noreply = cmd.no_reply;
```

Comment:
instead of setting a flag in side ConnectionState, why not pass  it to `mc_builder.` 
in mc_builder you can silently skip all replies if this flag is on.
After the dispatch command you can reset the flag in MCReplyBuilder.

This way your logic will reside in a concentrated place.